tests/MongoDB.Bson.Tests/Serialization/BsonDocumentBackedClassTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Reflection;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.TestHelpers;
+
+namespace MongoDB.Bson.Tests.Serialization
+{
+    public class BsonDocumentBackedClassTests
```

Comment:
We didn't have any tests for this before either. But this is the file were the BsonDocumentBackedClassReflector needs to live.

I can write tests for this abstract base class if you want. It's implicitly tested by the tests for ChangeStreamDocument.

Reply:
No need to explicitly test this with the implicit tests IMHO. Also, I'm curious: why does the extension method need to live in this class?

Reply:
Not this class... this file.

We've typically put the XyzReflector in the same file as XyzTests, because *usually* the primary consumer of the reflection methods are the XyzTests.

We *could* strictly adhere to the rule that every type live in a separate file, and then we'd have many more files than we do now.